### PR TITLE
Pydantic v2 upgrade

### PR DESCRIPTION
**Pull request by sblack-usu**
### Pull Request Checklist: 
- [ ] Positive Test Case Written by Dev

- [ ] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [ ] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. [Enter positive test case here]

### CODE REVIEW

File: hs_rest_api2/tests/test_file_json_metadata.py [hunk 1]
```diff
@@ -75,6 +75,7 @@ def compare_response(result_json, expected_json):
                 # overwrite system metadata fields for comparison
                 result_json['modified'] = expected_json['modified']
                 result_json['created'] = expected_json['created']
+                result_json['review_started'] = expected_json['review_started']
```
Comment (devincowan) [inline hs_rest_api2/tests/test_file_json_metadata.py:78]: thanks for adding this!

**Review by devincowan — APPROVED** [review]
thanks @sblack-usu looks good

File: Dockerfile [hunk 2]
```diff
@@ -7,6 +7,8 @@ RUN sed -i -e 's/# en_US.UTF-8 UTF-8/en_US.UTF-8 UTF-8/' /etc/locale.gen && \
 RUN pip install mozilla-django-oidc
 RUN pip install spam_patterns@git+https://github.com/CUAHSI/spam_patterns.git@0.0.4
 RUN pip install freezegun
+RUN pip install --upgrade hsmodels
```
Comment (devincowan) [inline Dockerfile:10]: when we release this, I would like to pin the hsmodels version in a newly minted image of hs_docker_base

File: Dockerfile [hunk 3]
```diff
@@ -7,6 +7,8 @@ RUN sed -i -e 's/# en_US.UTF-8 UTF-8/en_US.UTF-8 UTF-8/' /etc/locale.gen && \
...
+RUN pip install -U email-validator
```
Comment (devincowan) [inline Dockerfile:11]: why `email-validator` ?

Comment (sblack-usu) [reply Dockerfile:11]: I ran into a version mismatch with pydantic and an older version of email-validator. Updating email-validator resolved the issue.

File: Dockerfile (hunk 2 above)
Comment (sblack-usu) [reply Dockerfile:10]: Definitely. Are you planning on doing the same for the other pip installs in this file?

Comment (devincowan) [reply Dockerfile:10]: Yes. And ideally some of the other security issues we have
https://github.com/hydroshare/hs_docker_base/security/dependabot
Let's leave this PR open and once I get the base image for you, we can edit this Dockerfile and then merge?

**Review by devincowan — APPROVED** [review]

File: Dockerfile (hunk 2 above)
Comment (devincowan) [reply Dockerfile]: Okay I added all the security updates as well as new pinned hsmodels in the base image

**Pull request merged**